File: RoomTestJava/app/src/main/java/org/first/roomtestjava/MainActivity.java
package org.first.roomtestjava;

import androidx.appcompat.app.AppCompatActivity;
import androidx.room.Room;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView resultTextView;
    EditText todoEditText;
    Button addButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        resultTextView = findViewById(R.id.result_textview);
        todoEditText = findViewById(R.id.todo_edittext);
        addButton = findViewById(R.id.add_button);

        final AppDatabase db = Room.databaseBuilder(this, AppDatabase.class, "todo-db")
                .allowMainThreadQueries()
                .build();
        resultTextView.setText(db.todoDao().getAll().toString());
        addButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                db.todoDao().insert(new Todo(todoEditText.getText().toString()));
                resultTextView.setText(db.todoDao().getAll().toString());
            }
        });
    }
}

File: KotlinLiveDataTest/app/src/main/java/org/first/kotlinlivedatatest/MainActivity.kt
package org.first.kotlinlivedatatest

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import org.first.kotlinlivedatatest.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    var data = MutableLiveData<ArrayList<User>>()
    lateinit var adapter: RecyclerAdapter
    lateinit var binding : ActivityMainBinding
    lateinit var viewmodel : MainViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        viewmodel = ViewModelProviders.of(this).get(MainViewModel::class.java)



        binding.recycler.layoutManager = LinearLayoutManager(this)


        binding.funON.setOnClickListener {
            var list = ArrayList<User>()
            list.add( User("1","test1"))
            list.add( User("2","test2"))
            list.add( User("3","test3"))
            viewmodel.liveData.postValue(list)
        }

        val dataObserver: Observer<ArrayList<User>> =
            Observer { livedata ->
                data.value = livedata
                Log.e("dataValue1", data.value!![0].toString())
                Log.e("dataValue2", data.value!![1].toString())
                Log.e("dataValue3", data.value!![2].toString())
                var newAdapter = RecyclerAdapter(data)
                binding.recycler.adapter = newAdapter
            }

        viewmodel.liveData.observe(this, dataObserver)
    }
}